packages/task/src/node/process/process-task.ts

```diff
@@ -70,6 +72,25 @@ export class ProcessTask extends Task {
         });
     }
 
+    initClientConnection(connection: MessageConnection): void {
+        const process = this.options.process;
+        if (process instanceof RawProcess) {
+            const outputReadline = readline.createInterface({ input: process.output, crlfDelay: Infinity });
+            outputReadline.on('line', (data: string) => connection.sendNotification('onLine', data));
+
+            connection.onRequest('kill', (data: string) => process.kill());
+
+            connection.sendNotification('onStart');
+            process.onExit(event => connection.sendNotification('onExit', event));
+
+            connection.listen();
+        } else {
+            // We don't connect to terminal process here.  No tasks use terminal process
+            // and one would need a custom runner and a custom Task implementation if a task
+            // did use a terminal process.
```

Comment:
Is that true?

See: https://github.com/theia-ide/theia/blob/master/packages/task/src/node/process/process-task-runner.ts#L68-L89.

Reply:
Yes, sorry, I did write that comment a bit wrong.  What I really meant was that no tasks would connect to the terminal widget, as opposed to running as a shell.  We could provide a default implementation here for terminal processes too, though we don't currently have a need for that so perhaps we should not try to provide a default implementation for terminal processes.

Perhaps a better comment would be "No default implementation is provided to connect a terminal process to the client."